### ISSUE TITLE
When running tests in parallel detox does not launch and run on all emulators API 31+ some emulators timeout while setting up Detox environment using v20

### ISSUE THREAD

**Issue by tanakaderoy** [wix/Detox#3763]
### What happened?

 `--max-workers` flag not working correctly for Android 31 and above
 There was this fixed [issue](https://github.com/wix/Detox/issues/3342) that fixed detox not launching and connecting to android emulators API 31+ but it seems this fix did not propagate to the situation where we launch multiple emulators when running tests in parallel.
 
 if I run with `--max-workers 3` 1 emulator will launch and run tests just fine but then the other 2 will time out.

### What was the expected behaviour?

running tests with `-max-workers <n>` successfully launches and runs tests on <n> number of emulators  

### Was it tested on latest Detox?

- [X] I have tested this issue on the latest Detox release and it still reproduces.

### Did your test throw out a timeout?

- [ ] I have followed the instructions under [Identifying which synchronization mechanism causes us to wait too much](https://wix.github.io/Detox/docs/troubleshooting/synchronization#identifying-which-synchronization-mechanism-causes-us-to-wait-too-much).

### Help us reproduce this issue!

Cant get the DEtoxTemplate project to yarn install bbut i have a repro repo

https://github.com/tanakaderoy/androidapi31parallelrepro/

### In what environment did this happen?

Detox version: 20.1.0
React Native version:0.69/7 
Node version: 16.14.0
Device model: Pixel_API_31_AOSP
Android version: API 31
Test-runner (select one): jest 

### Detox logs

<details><summary>Detox logs</summary>

```
{"ph":"i","name":"dispatching event to  primary-89213 /tmp/detox.primary-89213  :  reportTestResults , {\n  testResults: [\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel3.e2e.js',\n      testExecError: undefined,\n      isPermanentFailure: false\n    },\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel1.e2e.js',\n      testExecError: undefined,\n      isPermanentFailure: false\n    },\n    {\n      success: false,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel2.e2e.js',\n      testExecError: {\n        '0': 'E',\n        '1': 'x',\n        '2': 'c',\n        '3': 'e',\n        '4': 'e',\n        '5': 'd',\n        '6': 'e',\n        '7': 'd',\n        '8': ' ',\n        '9': 't',\n        '10': 'i',\n        '11': 'm',\n        '12': 'e',\n        '13': 'o',\n        '14': 'u',\n        '15': 't',\n        '16': ' ',\n        '17': 'o',\n        '18': 'f',\n        '19': ' ',\n        '20': '1',\n        '21': '2',\n        '22': '0',\n        '23': '0',\n        '24': '0',\n        '25': '0',\n        '26': 'm',\n        '27': 's',\n        '28': ' ',\n        '29': 'w',\n        '30': 'h',\n        '31': 'i',\n        '32': 'l',\n        '33': 'e',\n        '34': ' ',\n        '35': 's',\n        '36': 'e',\n        '37': 't',\n        '38': 't',\n        '39': 'i',\n        '40': 'n',\n        '41': 'g',\n        '42': ' ',\n        '43': 'u',\n        '44': 'p',\n        '45': ' ',\n        '46': 'D',\n        '47': 'e',\n        '48': 't',\n        '49': 'o',\n        '50': 'x',\n        '51': ' ',\n        '52': 'e',\n        '53': 'n',\n        '54': 'v',\n        '55': 'i',\n        '56': 'r',\n        '57': 'o',\n        '58': 'n',\n        '59': 'm',\n        '60': 'e',\n        '61': 'n',\n        '62': 't',\n        type: 'Error',\n        stack: '',\n        message: 'Exceeded timeout of 120000ms while setting up Detox environment'\n      },\n      isPermanentFailure: false\n    }\n  ]\n}","pid":89214,"tid":6,"cat":"ipc","ts":1671230329380000,"args":{"level":10,"v":0}},
	{"ph":"i","name":"received event of :  reportTestResults {\n  testResults: [\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel3.e2e.js',\n      isPermanentFailure: false\n    },\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel1.e2e.js',\n      isPermanentFailure: false\n    },\n    {\n      success: false,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel2.e2e.js',\n      testExecError: {\n        '0': 'E',\n        '1': 'x',\n        '2': 'c',\n        '3': 'e',\n        '4': 'e',\n        '5': 'd',\n        '6': 'e',\n        '7': 'd',\n        '8': ' ',\n        '9': 't',\n        '10': 'i',\n        '11': 'm',\n        '12': 'e',\n        '13': 'o',\n        '14': 'u',\n        '15': 't',\n        '16': ' ',\n        '17': 'o',\n        '18': 'f',\n        '19': ' ',\n        '20': '1',\n        '21': '2',\n        '22': '0',\n        '23': '0',\n        '24': '0',\n        '25': '0',\n        '26': 'm',\n        '27': 's',\n        '28': ' ',\n        '29': 'w',\n        '30': 'h',\n        '31': 'i',\n        '32': 'l',\n        '33': 'e',\n        '34': ' ',\n        '35': 's',\n        '36': 'e',\n        '37': 't',\n        '38': 't',\n        '39': 'i',\n        '40': 'n',\n        '41': 'g',\n        '42': ' ',\n        '43': 'u',\n        '44': 'p',\n        '45': ' ',\n        '46': 'D',\n        '47': 'e',\n        '48': 't',\n        '49': 'o',\n        '50': 'x',\n        '51': ' ',\n        '52': 'e',\n        '53': 'n',\n        '54': 'v',\n        '55': 'i',\n        '56': 'r',\n        '57': 'o',\n        '58': 'n',\n        '59': 'm',\n        '60': 'e',\n        '61': 'n',\n        '62': 't',\n        type: 'Error',\n        stack: '',\n        message: 'Exceeded timeout of 120000ms while setting up Detox environment'\n      },\n      isPermanentFailure: false\n    }\n  ]\n}","pid":89213,"tid":1,"cat":"ipc,ipc-server","ts":1671230329384000,"args":{"level":10,"v":0}},
	{"ph":"i","name":"dispatching event to socket  :  reportTestResultsDone {\n  testResults: [\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel3.e2e.js',\n      isPermanentFailure: false\n    },\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel1.e2e.js',\n      isPermanentFailure: false\n    },\n    {\n      success: false,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel2.e2e.js',\n      testExecError: {\n        '0': 'E',\n        '1': 'x',\n        '2': 'c',\n        '3': 'e',\n        '4': 'e',\n        '5': 'd',\n        '6': 'e',\n        '7': 'd',\n        '8': ' ',\n        '9': 't',\n        '10': 'i',\n        '11': 'm',\n        '12': 'e',\n        '13': 'o',\n        '14': 'u',\n        '15': 't',\n        '16': ' ',\n        '17': 'o',\n        '18': 'f',\n        '19': ' ',\n        '20': '1',\n        '21': '2',\n        '22': '0',\n        '23': '0',\n        '24': '0',\n        '25': '0',\n        '26': 'm',\n        '27': 's',\n        '28': ' ',\n        '29': 'w',\n        '30': 'h',\n        '31': 'i',\n        '32': 'l',\n        '33': 'e',\n        '34': ' ',\n        '35': 's',\n        '36': 'e',\n        '37': 't',\n        '38': 't',\n        '39': 'i',\n        '40': 'n',\n        '41': 'g',\n        '42': ' ',\n        '43': 'u',\n        '44': 'p',\n        '45': ' ',\n        '46': 'D',\n        '47': 'e',\n        '48': 't',\n        '49': 'o',\n        '50': 'x',\n        '51': ' ',\n        '52': 'e',\n        '53': 'n',\n        '54': 'v',\n        '55': 'i',\n        '56': 'r',\n        '57': 'o',\n        '58': 'n',\n        '59': 'm',\n        '60': 'e',\n        '61': 'n',\n        '62': 't',\n        type: 'Error',\n        stack: '',\n        message: 'Exceeded timeout of 120000ms while setting up Detox environment'\n      },\n      isPermanentFailure: false\n    }\n  ]\n}","pid":89213,"tid":1,"cat":"ipc,ipc-server","ts":1671230329387000,"args":{"level":10,"v":0}},
	{"ph":"i","name":"broadcasting event to all known sockets listening to  /tmp/detox.primary-89213  :   sessionStateUpdate {\n  testResults: [\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel3.e2e.js',\n      isPermanentFailure: false\n    },\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel1.e2e.js',\n      isPermanentFailure: false\n    },\n    {\n      success: false,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel2.e2e.js',\n      testExecError: {\n        '0': 'E',\n        '1': 'x',\n        '2': 'c',\n        '3': 'e',\n        '4': 'e',\n        '5': 'd',\n        '6': 'e',\n        '7': 'd',\n        '8': ' ',\n        '9': 't',\n        '10': 'i',\n        '11': 'm',\n        '12': 'e',\n        '13': 'o',\n        '14': 'u',\n        '15': 't',\n        '16': ' ',\n        '17': 'o',\n        '18': 'f',\n        '19': ' ',\n        '20': '1',\n        '21': '2',\n        '22': '0',\n        '23': '0',\n        '24': '0',\n        '25': '0',\n        '26': 'm',\n        '27': 's',\n        '28': ' ',\n        '29': 'w',\n        '30': 'h',\n        '31': 'i',\n        '32': 'l',\n        '33': 'e',\n        '34': ' ',\n        '35': 's',\n        '36': 'e',\n        '37': 't',\n        '38': 't',\n        '39': 'i',\n        '40': 'n',\n        '41': 'g',\n        '42': ' ',\n        '43': 'u',\n        '44': 'p',\n        '45': ' ',\n        '46': 'D',\n        '47': 'e',\n        '48': 't',\n        '49': 'o',\n        '50': 'x',\n        '51': ' ',\n        '52': 'e',\n        '53': 'n',\n        '54': 'v',\n        '55': 'i',\n        '56': 'r',\n        '57': 'o',\n        '58': 'n',\n        '59': 'm',\n        '60': 'e',\n        '61': 'n',\n        '62': 't',\n        type: 'Error',\n        stack: '',\n        message: 'Exceeded timeout of 120000ms while setting up Detox environment'\n      },\n      isPermanentFailure: false\n    }\n  ]\n}","pid":89213,"tid":1,"cat":"ipc,ipc-server","ts":1671230329388000,"args":{"level":10,"v":0}},
	{"ph":"i","name":"## received events ##","pid":89214,"tid":6,"cat":"ipc","ts":1671230329388000,"args":{"level":10,"v":0}},
	{"ph":"i","name":"detected event reportTestResultsDone {\n  testResults: [\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel3.e2e.js',\n      isPermanentFailure: false\n    },\n    {\n      success: true,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel1.e2e.js',\n      isPermanentFailure: false\n    },\n    {\n      success: false,\n      testFilePath: '/Users/tanaka.mazivanhanga/Documents/GitHub/androidAPI31ParallelRepro/e2e/parallel2.e2e.js',\n      testExecError: {\n        '0': 'E',\n        '1': 'x',\n        '2': 'c',\n        '3': 'e',\n        '4': 'e',\n        '5': 'd',\n        '6': 'e',\n        '7': 'd',\n        '8': ' ',\n        '9': 't',\n        '10': 'i',\n        '11': 'm',\n        '12': 'e',\n        '13': 'o',\n        '14': 'u',\n        '15': 't',\n        '16': ' ',\n        '17': 'o',\n        '18': 'f',\n        '19': ' ',\n        '20': '1',\n        '21': '2',\n        '22': '0',\n        '23': '0',\n        '24': '0',\n        '25': '0',\n        '26': 'm',\n        '27': 's',\n        '28': ' ',\n        '29': 'w',\n        '30': 'h',\n        '31': 'i',\n        '32': 'l',\n        '33': 'e',\n        '34': ' ',\n        '35': 's',\n        '36': 'e',\n        '37': 't',\n        '38': 't',\n        '39': 'i',\n        '40': 'n',\n        '41': 'g',\n        '42': ' ',\n        '43': 'u',\n        '44': 'p',\n        '45': ' ',\n        '46': 'D',\n        '47': 'e',\n        '48': 't',\n        '49': 'o',\n        '50': 'x',\n        '51': ' ',\n        '52': 'e',\n        '53': 'n',\n        '54': 'v',\n        '55': 'i',\n        '56': 'r',\n        '57': 'o',\n        '58': 'n',\n        '59': 'm',\n        '60': 'e',\n        '61': 'n',\n        '62': 't',\n        type: 'Error',\n        stack: '',\n        message: 'Exceeded timeout of 120000ms while setting up Detox environment'\n      },\n      isPermanentFailure: false\n    }\n  ]\n}","pid":89214,"tid":6,"cat":"ipc","ts":1671230329389000,"args":{"level":10,"v":0}},
	{"ph":"i","name":"socket disconnected secondary-89214","pid":89213,"tid":1,"cat":"ipc,ipc-server","ts":1671230329392000,"args":{"level":10,"v":0}},
	{"ph":"i","name":"connection closed primary-89213 /tmp/detox.primary-89213 0 tries remaining of 0","pid":89214,"tid":6,"cat":"ipc","ts":1671230329392000,"args":{"level":10,"v":0}},
	{"ph":"i","name":"secondary-89214 exceeded connection rety amount of  or stopRetrying flag set.","pid":89214,"tid":6,"cat":"ipc","ts":1671230329392000,"args":{"level":10,"v":0}},
	{"ph":"E","pid":89213,"tid":0,"cat":"lifecycle,cli","ts":1671230329466000,"args":{"level":50,"success":false,"code":1,"signal":null,"v":0}},
	{"ph":"i","name":"Detox server has been closed gracefully","pid":89213,"tid":2,"cat":"ws-server,ws","ts":1671230329467000,"args":{"level":20,"v":0}},
	{"ph":"E","pid":89213,"tid":0,"cat":"lifecycle","ts":1671230329469000,"args":{"level":10,"v":0}}
```

</details>

### Device logs

<details><summary>Device logs</summary>

```
no failure logs
```

</details>

### More data, please!

![Screenshot 2022-12-16 at 5 41 42 PM](https://user-images.githubusercontent.com/42681052/208200684-512b0813-9213-4a23-94f9-e27d2259ca1d.png)

**Comment by noomorph** [comment]
Hi, the report looks good. I'll look on Monday-Tuesday

сб, 17 груд. 2022 р. о 00:42 Tanaka Mazivanhanga ***@***.***>
пише:

**Comment by noomorph** [comment]
<img width="1120" alt="image" src="https://user-images.githubusercontent.com/1962469/208383922-66cd6034-817c-4b8b-a981-db6a58a1bbcf.png">

@d4vidi @jonathanmos I think I see something like a reaction to stress load: e.g., 2 of 3 emulators boot, and 1 does not. Sometimes only one emulator boots.

The unlucky devices stay forever with a black screen.


<summary>

Here are some logs:

<details>

```
INFO    | Android emulator version 31.3.12.0 (build_id 9126400) (CL:N/A)
INFO    | Found AVD name 'Pixel_API_31_AOSP'
INFO    | Found AVD target architecture: x86_64
INFO    | argv[0]: '/Users/yaroslavs/Library/Android/sdk/emulator/emulator'; program directory: '/Users/yaroslavs/Library/Android/sdk/emulator'
VERBOSE |  Found directory: /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64/
emulator: INFO: Found systemPath /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64/
WARNING | unexpected system image feature string, emulator might not function correctly, please try updating the emulator.
WARNING | cannot add library /Users/yaroslavs/Library/Android/sdk/emulator/qemu/darwin-x86_64/lib64/vulkan/libvulkan.dylib: failed
INFO    | Duplicate loglines will be removed, if you wish to see each indiviudal line launch with the -log-nofilter flag.
VERBOSE | autoconfig: -skin pixel_silver
VERBOSE | autoconfig: -skindir /Users/yaroslavs/Library/Android/sdk/skins/
VERBOSE | Unexpected feature list:
VERBOSE | Vulkan
VERBOSE | autoconfig: -kernel /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//kernel-ranchu
VERBOSE | Target arch = 'x86_64'
VERBOSE | Auto-detect: Kernel image requires new device naming scheme.
VERBOSE | Auto-detect: Kernel does not support YAFFS2 partitions.
VERBOSE | autoconfig: -ramdisk /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//ramdisk.img
VERBOSE | Using initial system image: /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//system.img
VERBOSE | Using initial vendor image: /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//vendor.img
VERBOSE | autoconfig: -data /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/userdata-qemu.img
VERBOSE | autoconfig: -initdata /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/userdata.img
VERBOSE | autoconfig: -cache /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/cache.img
VERBOSE | autoconfig: -sdcard /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/sdcard.img
VERBOSE | Increasing RAM size to 2048MB
VERBOSE | VM heap size 228MB is below hardware specified minimum of 512MB,setting it to that value
VERBOSE | System image is read only
VERBOSE | Found 2 DNS servers:
VERBOSE | 	8.8.8.8
VERBOSE | 	1.1.1.1
VERBOSE | trying to load skin file '/Users/yaroslavs/Library/Android/sdk/skins//pixel_silver/layout'
VERBOSE | Requested console port 14328: Inferring adb port 14329.
VERBOSE | CPU Acceleration: working
VERBOSE | CPU Acceleration status: Hypervisor.Framework OS X Version 13.0
VERBOSE | handleCpuAcceleration: feature check for hvf
VERBOSE | Bluetooth requested by guest
VERBOSE | started modem simulator host server at port: 55658
VERBOSE | GPU emulation enabled using 'host' mode
VERBOSE | Initializing hardware OpenGLES emulation support
VERBOSE | create display 0
VERBOSE | setDisplayPose 0 x 0 y 0 w 1080 h 1920 dpi 0
INFO    | configAndStartRenderer: setting vsync to 60 hz
INFO    | Sending adb public key [QAAAAEfs75aJAi5z8kUs3acll+jJ/x5upSJAyv/Bp8UQPKhg6Hwq793r88Vr8A0ft81uEoz/y9NhHZJ7U9RsFkwfkJyAljSIxOoLJ86AHYYilYaEyuNTSv0l55E8NS8INKT3pCNeC6+bvhUtGdqSZTnMib0TU9wFyLg8LsTvnwDjDJClVnjcal0WJOS/C8gw5N40guWbi8+SFc23TdHhrPXpz/z8URrL88qoye0+j3smugsbKkeC6n9LYmP/nFqhUMuacAU1CQsVOmZPWDbuBW7l3kH/hb4QGEFUmSsFj2vAEzrOYA+6yxX+mKzHCkhZs2Nuo9k9BiMxhvVfjZvPKJtnvDMbu4jn6gYT6LazWEBpMqNu4Z+ZGskT5J+hkof5khJ6cmeajXpJCrp0nEqrpkBGUoQbpD6Jkxp6iKC/uiboV7D5Ii7XFtZQwxz4YBkf591RhUgM7ORT+/n+jz2gFSyBy7l3UIwlVvQYm+Qndv4JuUWy5F4+iwI9c3T15EKsLLTlAB4gMTdp2K1yznKPdnCpehXkJ1ZzQmvvgo3TGQOaaMtD1kGLCN44+8K14qHhZmeg5jFSP538w+DcFyH7jeFKxCWdSQXlhaTmwHe/64svG1c4wxlNNrKwT96Eh5ujqYSZLD2V5os+bciPFFnKTPRklzqjChe60PvoKLRvYaAF0nTM5eb9YgEAAQA= yaroslavs@unknown]
INFO    | added library /Users/yaroslavs/Library/Android/sdk/emulator/lib64/vulkan/libvulkan.dylib
VERBOSE | Found 2 DNS servers:
VERBOSE | 	8.8.8.8
VERBOSE | 	1.1.1.1
INFO    | Content of hardware configuration file:
INFO    | 	hw.cpu.arch = x86_64
INFO    | 	hw.cpu.ncore = 4
INFO    | 	hw.ramSize = 2048
INFO    | 	hw.screen = multi-touch
INFO    | 	hw.mainKeys = false
INFO    | 	hw.trackBall = false
INFO    | 	hw.keyboard = true
INFO    | 	hw.keyboard.lid = false
INFO    | 	hw.keyboard.charmap = qwerty2
INFO    | 	hw.dPad = false
INFO    | 	hw.rotaryInput = false
INFO    | 	hw.gsmModem = true
INFO    | 	hw.gps = true
INFO    | 	hw.battery = true
INFO    | 	hw.accelerometer = true
INFO    | 	hw.gyroscope = true
INFO    | 	hw.audioInput = true
INFO    | 	hw.audioOutput = true
INFO    | 	hw.sdCard = true
INFO    | 	hw.sdCard.path = /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/sdcard.img
INFO    | 	disk.cachePartition = true
INFO    | 	disk.cachePartition.path = /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/cache.img
INFO    | 	disk.cachePartition.size = 66m
INFO    | 	test.quitAfterBootTimeOut = -1
INFO    | 	test.delayAdbTillBootComplete = 0
INFO    | 	test.monitorAdb = 0
INFO    | 	hw.lcd.width = 1080
INFO    | 	hw.lcd.height = 1920
INFO    | 	hw.lcd.depth = 16
INFO    | 	hw.lcd.density = 420
INFO    | 	hw.lcd.backlight = true
INFO    | 	hw.lcd.vsync = 60
INFO    | 	hw.gltransport = pipe
INFO    | 	hw.gltransport.asg.writeBufferSize = 1048576
INFO    | 	hw.gltransport.asg.writeStepSize = 4096
INFO    | 	hw.gltransport.asg.dataRingSize = 32768
INFO    | 	hw.gltransport.drawFlushInterval = 800
INFO    | 	hw.displayRegion.0.1.xOffset = -1
INFO    | 	hw.displayRegion.0.1.yOffset = -1
INFO    | 	hw.displayRegion.0.1.width = 0
INFO    | 	hw.displayRegion.0.1.height = 0
INFO    | 	hw.displayRegion.0.2.xOffset = -1
INFO    | 	hw.displayRegion.0.2.yOffset = -1
INFO    | 	hw.displayRegion.0.2.width = 0
INFO    | 	hw.displayRegion.0.2.height = 0
INFO    | 	hw.displayRegion.0.3.xOffset = -1
INFO    | 	hw.displayRegion.0.3.yOffset = -1
INFO    | 	hw.displayRegion.0.3.width = 0
INFO    | 	hw.displayRegion.0.3.height = 0
INFO    | 	hw.display1.width = 0
INFO    | 	hw.display1.height = 0
INFO    | 	hw.display1.density = 0
INFO    | 	hw.display1.xOffset = -1
INFO    | 	hw.display1.yOffset = -1
INFO    | 	hw.display1.flag = 0
INFO    | 	hw.display2.width = 0
INFO    | 	hw.display2.height = 0
INFO    | 	hw.display2.density = 0
INFO    | 	hw.display2.xOffset = -1
INFO    | 	hw.display2.yOffset = -1
INFO    | 	hw.display2.flag = 0
INFO    | 	hw.display3.width = 0
INFO    | 	hw.display3.height = 0
INFO    | 	hw.display3.density = 0
INFO    | 	hw.display3.xOffset = -1
INFO    | 	hw.display3.yOffset = -1
INFO    | 	hw.display3.flag = 0
INFO    | 	hw.multi_display_window = false
INFO    | 	hw.gpu.enabled = true
INFO    | 	hw.gpu.mode = host
INFO    | 	hw.initialOrientation = Portrait
INFO    | 	hw.camera.back = virtualscene
INFO    | 	hw.camera.front = emulated
INFO    | 	vm.heapSize = 512
INFO    | 	hw.sensors.light = true
INFO    | 	hw.sensors.pressure = true
INFO    | 	hw.sensors.humidity = true
INFO    | 	hw.sensors.proximity = true
INFO    | 	hw.sensors.magnetic_field = true
INFO    | 	hw.sensors.magnetic_field_uncalibrated = true
INFO    | 	hw.sensors.gyroscope_uncalibrated = true
INFO    | 	hw.sensors.orientation = true
INFO    | 	hw.sensors.temperature = true
INFO    | 	hw.sensors.rgbclight = false
INFO    | 	hw.sensor.hinge = false
INFO    | 	hw.sensor.hinge.count = 0
INFO    | 	hw.sensor.hinge.type = 0
INFO    | 	hw.sensor.hinge.sub_type = 0
INFO    | 	hw.sensor.hinge.fold_to_displayRegion.0.1_at_posture = 1
INFO    | 	hw.sensor.roll = false
INFO    | 	hw.sensor.roll.count = 0
INFO    | 	hw.sensor.roll.resize_to_displayRegion.0.1_at_posture = 6
INFO    | 	hw.sensor.roll.resize_to_displayRegion.0.2_at_posture = 6
INFO    | 	hw.sensor.roll.resize_to_displayRegion.0.3_at_posture = 6
INFO    | 	hw.sensors.heart_rate = false
INFO    | 	hw.sensors.wrist_tilt = false
INFO    | 	hw.useext4 = true
INFO    | 	hw.arc = false
INFO    | 	hw.arc.autologin = false
INFO    | 	hw.device.name = pixel
INFO    | 	kernel.path = /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//kernel-ranchu
INFO    | 	kernel.newDeviceNaming = yes
INFO    | 	kernel.supportsYaffs2 = no
INFO    | 	disk.ramdisk.path = /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//ramdisk.img
INFO    | 	disk.systemPartition.initPath = /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//system.img
INFO    | 	disk.systemPartition.size = 4107m
INFO    | 	disk.vendorPartition.initPath = /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//vendor.img
INFO    | 	disk.vendorPartition.size = 800m
INFO    | 	disk.dataPartition.path = /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/userdata-qemu.img
INFO    | 	disk.dataPartition.size = 6g
INFO    | 	disk.encryptionKeyPartition.path = /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/encryptionkey.img
INFO    | 	PlayStore.enabled = false
INFO    | 	avd.name = Pixel_API_31_AOSP
INFO    | 	avd.id = Pixel_API_31_AOSP
INFO    | 	fastboot.forceColdBoot = false
INFO    | 	android.sdk.root = /Users/yaroslavs/Library/Android/sdk
INFO    | 	android.avd.home = /Users/yaroslavs/.android/avd
INFO    | .
INFO    | QEMU options list:
INFO    | 	 argv[00] = "/Users/yaroslavs/Library/Android/sdk/emulator/qemu/darwin-x86_64/qemu-system-x86_64"
INFO    | 	 argv[01] = "-read-only"
INFO    | 	 argv[02] = "-dns-server"
INFO    | 	 argv[03] = "8.8.8.8,1.1.1.1"
INFO    | 	 argv[04] = "-no-audio"
INFO    | 	 argv[05] = "-mem-path"
INFO    | 	 argv[06] = "/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/snapshots/default_boot/ram.img"
INFO    | 	 argv[07] = "-serial"
INFO    | 	 argv[08] = "null"
INFO    | 	 argv[09] = "-android-ports"
INFO    | 	 argv[10] = "14328,14329"
INFO    | 	 argv[11] = "-device"
INFO    | 	 argv[12] = "goldfish_pstore,addr=0xff018000,size=0x10000,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/data/misc/pstore/pstore.bin"
INFO    | 	 argv[13] = "-cpu"
INFO    | 	 argv[14] = "android64"
INFO    | 	 argv[15] = "-enable-hvf"
INFO    | 	 argv[16] = "-smp"
INFO    | 	 argv[17] = "cores=4"
INFO    | 	 argv[18] = "-m"
INFO    | 	 argv[19] = "2048"
INFO    | 	 argv[20] = "-lcd-density"
INFO    | 	 argv[21] = "420"
INFO    | 	 argv[22] = "-nodefaults"
INFO    | 	 argv[23] = "-kernel"
INFO    | 	 argv[24] = "/Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//kernel-ranchu"
INFO    | 	 argv[25] = "-initrd"
INFO    | 	 argv[26] = "/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/initrd"
INFO    | 	 argv[27] = "-drive"
INFO    | 	 argv[28] = "if=none,index=0,id=system,if=none,file=/Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//system.img,read-only"
INFO    | 	 argv[29] = "-device"
INFO    | 	 argv[30] = "virtio-blk-pci,drive=system,modern-pio-notify"
INFO    | 	 argv[31] = "-drive"
INFO    | 	 argv[32] = "if=none,index=1,id=cache,if=none,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/cache.img.qcow2,overlap-check=none,cache=unsafe,l2-cache-size=1048576"
INFO    | 	 argv[33] = "-device"
INFO    | 	 argv[34] = "virtio-blk-pci,drive=cache,modern-pio-notify"
INFO    | 	 argv[35] = "-drive"
INFO    | 	 argv[36] = "if=none,index=2,id=userdata,if=none,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/userdata-qemu.img.qcow2,overlap-check=none,cache=unsafe,l2-cache-size=1048576"
INFO    | 	 argv[37] = "-device"
INFO    | 	 argv[38] = "virtio-blk-pci,drive=userdata,modern-pio-notify"
INFO    | 	 argv[39] = "-drive"
INFO    | 	 argv[40] = "if=none,index=3,id=encrypt,if=none,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/encryptionkey.img.qcow2,overlap-check=none,cache=unsafe,l2-cache-size=1048576"
INFO    | 	 argv[41] = "-device"
INFO    | 	 argv[42] = "virtio-blk-pci,drive=encrypt,modern-pio-notify"
INFO    | 	 argv[43] = "-drive"
INFO    | 	 argv[44] = "if=none,index=4,id=vendor,if=none,file=/Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//vendor.img,read-only"
INFO    | 	 argv[45] = "-device"
INFO    | 	 argv[46] = "virtio-blk-pci,drive=vendor,modern-pio-notify"
INFO    | 	 argv[47] = "-drive"
INFO    | 	 argv[48] = "if=none,index=5,id=sdcard,if=none,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/sdcard.img.qcow2,overlap-check=none,cache=unsafe,l2-cache-size=1048576"
INFO    | 	 argv[49] = "-device"
INFO    | 	 argv[50] = "virtio-blk-pci,drive=sdcard,modern-pio-notify"
INFO    | 	 argv[51] = "-netdev"
INFO    | 	 argv[52] = "user,id=mynet"
INFO    | 	 argv[53] = "-device"
INFO    | 	 argv[54] = "virtio-net-pci,netdev=mynet"
INFO    | 	 argv[55] = "-chardev"
INFO    | 	 argv[56] = "null,id=forhvc0"
INFO    | 	 argv[57] = "-chardev"
INFO    | 	 argv[58] = "null,id=forhvc1"
INFO    | 	 argv[59] = "-device"
INFO    | 	 argv[60] = "virtio-serial-pci,ioeventfd=off"
INFO    | 	 argv[61] = "-device"
INFO    | 	 argv[62] = "virtconsole,chardev=forhvc0"
INFO    | 	 argv[63] = "-device"
INFO    | 	 argv[64] = "virtconsole,chardev=forhvc1"
INFO    | 	 argv[65] = "-chardev"
INFO    | 	 argv[66] = "rootcanal,id=rootcanal"
INFO    | 	 argv[67] = "-device"
INFO    | 	 argv[68] = "virtserialport,chardev=rootcanal,name=bluetooth"
INFO    | 	 argv[69] = "-device"
INFO    | 	 argv[70] = "virtio-serial,ioeventfd=off"
INFO    | 	 argv[71] = "-chardev"
INFO    | 	 argv[72] = "socket,port=55657,host=::1,nowait,nodelay,ipv6,id=modem"
INFO    | 	 argv[73] = "-device"
INFO    | 	 argv[74] = "virtserialport,chardev=modem,name=modem"
INFO    | 	 argv[75] = "-device"
INFO    | 	 argv[76] = "virtio-rng-pci"
INFO    | 	 argv[77] = "-show-cursor"
INFO    | 	 argv[78] = "-device"
INFO    | 	 argv[79] = "virtio_input_multi_touch_pci_1"
INFO    | 	 argv[80] = "-device"
INFO    | 	 argv[81] = "virtio_input_multi_touch_pci_2"
INFO    | 	 argv[82] = "-device"
INFO    | 	 argv[83] = "virtio_input_multi_touch_pci_3"
INFO    | 	 argv[84] = "-device"
INFO    | 	 argv[85] = "virtio_input_multi_touch_pci_4"
INFO    | 	 argv[86] = "-device"
INFO    | 	 argv[87] = "virtio_input_multi_touch_pci_5"
INFO    | 	 argv[88] = "-device"
INFO    | 	 argv[89] = "virtio_input_multi_touch_pci_6"
INFO    | 	 argv[90] = "-device"
INFO    | 	 argv[91] = "virtio_input_multi_touch_pci_7"
INFO    | 	 argv[92] = "-device"
INFO    | 	 argv[93] = "virtio_input_multi_touch_pci_8"
INFO    | 	 argv[94] = "-device"
INFO    | 	 argv[95] = "virtio_input_multi_touch_pci_9"
INFO    | 	 argv[96] = "-device"
INFO    | 	 argv[97] = "virtio_input_multi_touch_pci_10"
INFO    | 	 argv[98] = "-device"
INFO    | 	 argv[99] = "virtio_input_multi_touch_pci_11"
INFO    | 	 argv[100] = "-device"
INFO    | 	 argv[101] = "virtio-keyboard-pci"
INFO    | 	 argv[102] = "-netdev"
INFO    | 	 argv[103] = "user,id=virtio-wifi,dhcpstart=10.0.2.16"
INFO    | 	 argv[104] = "-device"
INFO    | 	 argv[105] = "virtio-wifi-pci,netdev=virtio-wifi"
INFO    | 	 argv[106] = "-device"
INFO    | 	 argv[107] = "virtio-vsock-pci,guest-cid=77"
INFO    | 	 argv[108] = "-L"
INFO    | 	 argv[109] = "/Users/yaroslavs/Library/Android/sdk/emulator/lib/pc-bios"
INFO    | 	 argv[110] = "-soundhw"
INFO    | 	 argv[111] = "hda"
INFO    | 	 argv[112] = "-vga"
INFO    | 	 argv[113] = "none"
INFO    | 	 argv[114] = "-append"
INFO    | 	 argv[115] = "no_timer_check 8250.nr_uarts=1 clocksource=pit console=0 cma=288M@0-4G ndns=2 loop.max_part=7 ramoops.mem_address=0xff018000 ramoops.mem_size=0x10000 memmap=0x10000$0xff018000 printk.devkmsg=on bootconfig"
INFO    | 	 argv[116] = "-android-hw"
INFO    | 	 argv[117] = "/tmp/android-yaroslavs/emulator-VNnAc7"
INFO    | Concatenated QEMU options: /Users/yaroslavs/Library/Android/sdk/emulator/qemu/darwin-x86_64/qemu-system-x86_64 -read-only -dns-server 8.8.8.8,1.1.1.1 -no-audio -mem-path /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/snapshots/default_boot/ram.img -serial null -android-ports 14328,14329 -device goldfish_pstore,addr=0xff018000,size=0x10000,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/data/misc/pstore/pstore.bin -cpu android64 -enable-hvf -smp cores=4 -m 2048 -lcd-density 420 -nodefaults -kernel /Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//kernel-ranchu -initrd /Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/initrd -drive if=none,index=0,id=system,if=none,file=/Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//system.img,read-only -device virtio-blk-pci,drive=system,modern-pio-notify -drive if=none,index=1,id=cache,if=none,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/cache.img.qcow2,overlap-check=none,cache=unsafe,l2-cache-size=1048576 -device virtio-blk-pci,drive=cache,modern-pio-notify -drive if=none,index=2,id=userdata,if=none,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/userdata-qemu.img.qcow2,overlap-check=none,cache=unsafe,l2-cache-size=1048576 -device virtio-blk-pci,drive=userdata,modern-pio-notify -drive if=none,index=3,id=encrypt,if=none,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/encryptionkey.img.qcow2,overlap-check=none,cache=unsafe,l2-cache-size=1048576 -device virtio-blk-pci,drive=encrypt,modern-pio-notify -drive if=none,index=4,id=vendor,if=none,file=/Users/yaroslavs/Library/Android/sdk/system-images/android-31/default/x86_64//vendor.img,read-only -device virtio-blk-pci,drive=vendor,modern-pio-notify -drive if=none,index=5,id=sdcard,if=none,file=/Users/yaroslavs/.android/avd/Pixel_API_31_AOSP.avd/sdcard.img.qcow2,overlap-check=none,cache=unsafe,l2-cache-size=1048576 -device virtio-blk-pci,drive=sdcard,modern-pio-notify -netdev user,id=mynet -device virtio-net-pci,netdev=myn
VERBOSE | Starting hostapd main loop.
VERBOSE | Android qemu version 31.3.12.0 (build_id 9126400) (CL:N/A)
VERBOSE | Starting QEMU main loop
VERBOSE | registered 'boot-properties' qemud service
VERBOSE | Adding boot property: 'ro.opengles.version' = '131072'
VERBOSE | Adding boot property: 'qemu.sf.fake_camera' = 'front'
VERBOSE | Adding boot property: 'dalvik.vm.heapsize' = '512m'
VERBOSE | Adding boot property: 'qemu.hw.mainkeys' = '0'
VERBOSE | Adding boot property: 'qemu.sf.lcd_density' = '420'
INFO    | injectedQemuChannel!
INFO    | Informing listeners of injection.
INFO    | Rootcanal has  been activated.
VERBOSE | goldfish_events.have-dpad: false
VERBOSE | goldfish_events.have-trackball: false
VERBOSE | goldfish_events.have-camera: true
VERBOSE | goldfish_events.have-keyboard: false
VERBOSE | goldfish_events.have-lidswitch: false
VERBOSE | goldfish_events.have-tabletmode: false
VERBOSE | goldfish_events.have-touch: false
VERBOSE | goldfish_events.have-multitouch: false
VERBOSE | Adding boot property: 'net.wifi_mac_prefix' = '14328'
WARNING | Requested adb port (14329) is outside the recommended range [5555,5586]. ADB may not function properly for the emulator. See -help-port for details.
VERBOSE | Not using any http proxy
VERBOSE | Adding boot property: 'qemu.timezone' = 'Europe/Kiev'
VERBOSE | android_hw_fingerprint_init: fingerprint qemud listen service initialized
VERBOSE | emulator_window_fb_rotate
INFO    | setDisplayConfigs w 1080 h 1920 dpiX 420 dpiY 420
VERBOSE | config multidisplay with config.ini 0x0 0x0 0x0
VERBOSE | No acpi ini file provided, using default
VERBOSE | OpenGL Vendor=[Google (ATI Technologies Inc.)]
VERBOSE | OpenGL Renderer=[Android Emulator OpenGL ES Translator (AMD Radeon Pro 560 OpenGL Engine)]
VERBOSE | OpenGL Version=[OpenGL ES 3.0 (4.1 ATI-4.9.48)]
VERBOSE | Path:/Users/yaroslavs/Library/Android/sdk/platform-tools/adb protocol version: 41
VERBOSE | Found: 3 adb executables (2x)
VERBOSE | Found: 3 adb executables
VERBOSE | Adb: /Users/yaroslavs/Library/Android/sdk/platform-tools/adb
VERBOSE | Path:/Users/yaroslavs/Library/Android/sdk/platform-tools/adb protocol version: 41 (2x)
VERBOSE | Path:/Users/yaroslavs/Library/Android/sdk/platform-tools/adb protocol version: 41
VERBOSE | No acpi ini file provided, using default (2x)
VERBOSE | No acpi ini file provided, using default
```
</details>

</summary>

I don't see anything suspicious.

So, the only thing that could have helped probably is to kill & respawn here...... 🤔 Dunno, any thoughts?

**Comment by noomorph** [comment]
I think we also can reproduce it in a simpler way:

```sh
#!/usr/bin/env sh

$ANDROID_SDK_ROOT/emulator/emulator -verbose -no-audio -no-boot-anim -read-only -port 14328 @Pixel_API_31_AOSP & \
$ANDROID_SDK_ROOT/emulator/emulator -verbose -no-audio -no-boot-anim -read-only -port 14329 @Pixel_API_31_AOSP & \
$ANDROID_SDK_ROOT/emulator/emulator -verbose -no-audio -no-boot-anim -read-only -port 14330 @Pixel_API_31_AOSP & \
wait

```

So far, I see some weird messages (they are not tracked by Detox possibly):

```
[18849:113667:1219/105804.012059:ERROR:backend_impl.cc(1065)] Critical error found -8
[18849:113667:1219/105804.013642:ERROR:entry_impl.cc(1062)] No file for c1030092
[18849:113667:1219/105804.015843:ERROR:entry_impl.cc(1062)] No file for c103005e
[18850:105731:1219/105804.041854:ERROR:backend_impl.cc(1065)] Critical error found -8
[18850:105731:1219/105804.042005:ERROR:entry_impl.cc(1062)] No file for c1030004
[18850:105731:1219/105804.042055:ERROR:entry_impl.cc(1062)] No file for a1010046
[18850:105731:1219/105804.047047:ERROR:entry_impl.cc(1062)] No file for a3010048
[18850:105731:1219/105804.049417:ERROR:entry_impl.cc(1062)] No file for c103004a
[18850:105731:1219/105804.069318:ERROR:entry_impl.cc(1062)] No file for c1030004
[18850:105731:1219/105804.160001:ERROR:block_files.cc(471)] Failed to open /Users/yaroslavs/Library/Caches/Android Open Source Project/Emulator/QtWebEngine/Default/Cache/data_0
[18849:113667:1219/105805.842986:ERROR:backend_impl.cc(1065)] Critical error found -8
[18849:113667:1219/105805.898044:ERROR:entry_impl.cc(985)] Failed to save user data
[18849:113667:1219/105805.898078:ERROR:entry_impl.cc(985)] Failed to save user data
```

I think we could play with some pauses between launches. 🤔  Not sure if that would help though.

**Comment by noomorph** [comment]
Duplicate of #3763.
